static/index.js

```diff
@@ -0,0 +1,136 @@
+
+  const reviewElement = document.createElement("span");
+  reviewElement.textContent = review.review;
+  reviewRatingElement.appendChild(reviewElement);
+
+  const ratingElement = document.createElement("span");
+  ratingElement.textContent = review.rating + " / 5";
+  reviewRatingElement.appendChild(ratingElement);
+
+  const ratingStarsElement = document.createElement("div");
+  ratingStarsElement.id = "rating-stars";
+  for (let i = 1; i <= 5; i++) {
+    const ratingStar = document.createElement("span");
+    ratingStar.textContent = "☆";
+    if (i <= review.rating) {
+      ratingStar.style.color = "gold";
+    }
+    ratingStarsElement.appendChild(ratingStar);
+  }
+  reviewRatingElement.appendChild(ratingStarsElement);
+
+  const deleteButton = document.createElement("button");
+  deleteButton.textContent = "Remove";
+  deleteButton.onclick = () => {
+    reviewsList = reviewsList.filter((r) => {
+      return r !== review;
+    });
+    renderReviewsList();
+  };
+  reviewRatingElement.appendChild(deleteButton);
+
+  return reviewRatingElement;
+}
+
+const newReviewButtonHandler = () => {
+  const reviewInput = document.getElementById("review");
+  const ratingInputList = document.querySelectorAll(".star");
+  let ratingValue = 0;
+  ratingInputList.forEach((ratingInput) => {
+    if (ratingInput.style.color === "gold") {
+      ratingInput.style.color = "gray";
+      ratingValue += 1;
+    }
+  });
+
+  if (!reviewInput.value || !ratingValue) {
+    return;
+  }
+
+  const newReview = {
+    review: reviewInput.value,
+    rating: ratingValue,
+  };
+
+  reviewsList.push(newReview);
+  renderReviewsList();
+  reviewInput.value = "";
+};
+
+function calculateAverageRating() {
+  let averageRating = 0;
+  reviewsList.forEach((review) => {
+    averageRating += Number(review.rating);
+  });
+  if (!reviewsList.length) {
+    averageRating = 0;
+  } else {
+    averageRating /= reviewsList.length;
+    averageRating = averageRating.toPrecision(2);
+  }
+
+  const averageRatingElement = document.getElementById("average-rating");
+  if (averageRating > 0) {
+    if (Number(averageRating.toString()[2]) === 0) {
+      averageRatingElement.textContent = averageRating.toString()[0] + " / 5";
+    } else {
+      averageRatingElement.textContent = averageRating + " / 5";
+    }
+  } else {
+    averageRatingElement.textContent = "There are no ratings currently.";
+  }
+}
+
+const addRating = (id) => {
+  const value = id.split("-")[1];
+  const starLabel = document.getElementById(id);
+
+  if (starLabel.style.color === "gray" || starLabel.style.color === "") {
+    for (let i = 1; i <= value; i++) {
+      const starLabel = document.getElementById("star-" + String(i));
+      starLabel.style.color = "gold";
+    }
+  } else {
+    for (let i = value; i <= 5; i++) {
+      const starLabel = document.getElementById("star-" + String(i));
+      starLabel.style.color = "gray";
+    }
+  }
+};
+
+reviewsList = loadFromLocalStorage();
```

Comment:
Ukoliko u `localStorage`-u nemaš spremljenih reviewova funkcija `loadFromLocalStorage` će ti vratiti `null` i pregazit ćet inicijalni review koji imaš definiran gore.
U tom slučaju `reviewsList = null`. Kad pokušaš dodati prvi review (linija 89) aplikacija pukne jer pokusavas pozvati `.push` nad null a to nije moguće.